### PR TITLE
Replaces torch_utils `set_seed` method with IsaacLab implementation

### PR DESCRIPTION
**Pull request by pascal-roth**
# Description

Replace torch_utils `set_seed` method with IsaacLab implementation

## Type of change

- Dependency removal

## Checklist

- [ ] I have read and understood the [contribution guidelines](https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html)
- [ ] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [ ] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab/isaaclab/utils/seed.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def set_seed(seed: int, torch_deterministic: bool = False) -> int:
+    """set seed across modules"""
+    if seed == -1 and torch_deterministic:
+        seed = 42
+    elif seed == -1:
+        seed = np.random.randint(0, 10000)
+    print(f"Setting seed: {seed}")
+
```
Comment (Mayankm96) [inline source/isaaclab/isaaclab/utils/seed.py]:
```suggestion
    if seed == -1:
        seed = 42 if  torch_deterministic else np.random.randint(0, 10000)
```

File: source/isaaclab/isaaclab/utils/seed.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import numpy as np
+import os
+import random
+import torch
+
+import warp as wp
+
+
+def set_seed(seed: int, torch_deterministic: bool = False) -> int:
```
Comment (Mayankm96) [inline source/isaaclab/isaaclab/utils/seed.py]: NIT: To me a "set" usually should be combined with "get". Do you want to add something like "get_seed" as well? Or maybe we call this function `configure_seed`?

File: source/isaaclab/isaaclab/utils/seed.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    """set seed across modules"""
```
[the remark this answers is not in the document]
Comment (Mayankm96) [reply source/isaaclab/isaaclab/utils/seed.py]: Should seed also support `None` type which means it becomes "-1" behavior? I feel it is more inutitive than -1. Also are negative number generators valid? It seems ti is a yes for random and numpy at least.

File: source/isaaclab/isaaclab/utils/seed.py (hunk 2 above)
Comment (pascal-roth) [reply source/isaaclab/isaaclab/utils/seed.py]: In that case, I would rename it! Only effect is a bigger break as also the name of the function changes

**Comment by ooctipus** [comment]
`source/isaaclab_tasks/test/test_factory_environments.py` and `source/isaaclab_rl/test/test_rl_games_wrapper.py` did not time out locally.

**Comment by pascal-roth** [comment]
@kellyguo11 and @Mayankm96, how do we want to handle the license as the function is taken from IsaacSIm 

https://github.com/isaac-sim/IsaacSim/blob/main/source/extensions/isaacsim.core.utils/python/impl/torch/maths.py

File: source/isaaclab/isaaclab/utils/seed.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2022-2025, The Isaac Lab Project Developers (https://github.com/isaac-sim/IsaacLab/blob/main/CONTRIBUTORS.md).
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+# This method has been adjusted from
+# https://github.com/isaac-sim/IsaacSim/blob/main/source/extensions/isaacsim.core.utils/python/impl/torch/maths.py
+
```
Comment (pascal-roth) [inline source/isaaclab/isaaclab/utils/seed.py]:
```suggestion
```
